### ISSUE TITLE
YOLO COMMAND NOT FOUND

### ISSUE THREAD

**Issue by tanu-04** [ultralytics/ultralytics#3324]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I HAVE GIT CLONED INSTALLED ALL PRE REQUISITES BUT AN ERROR POPPED ON SAYING YOLO COMMAND NOT FOUND

**Comment by glenn-jocher** [comment]
@tanu-04 see https://docs.ultralytics.com/quickstart/ for installation instructions

**Comment by tanu-04** [comment]
Ohh thankyou it's working now ..

**Comment by pderrenger** [comment]
@tanu-04 glad to hear it's working now! If you have any more questions or run into any issues, feel free to ask. 😊

**Comment by meeselizabeth** [comment]
Hi, I am trying to train yolov8 through SSH so that I can train it on my server, but I keep getting error "command "yolo" not found". Do you know a way to solve this?

**Comment by glenn-jocher** [comment]
@meeselizabeth hello! It sounds like the `yolo` command-line tool is not recognized in your server's environment. To ensure `yolo` commands work through SSH on your server, you need to run them from within the root directory of your cloned YOLOv8 repository or ideally, you should have the `yolo` command-line tool available in your `PATH`.

Here are a few steps to check and solve this:

1. **Navigate to the YOLOv8 directory**: Make sure you are in the root directory of the cloned YOLOv8 repo.

```bash
cd /path/to/your/yolov8/
```

2. **Activate the environment**: If you are using a virtual environment (e.g., conda or venv), ensure that it is activated.

3. **Install package (if not done already)**: If you haven't already installed YOLOv8 as a package, you can do so by running the `pip install` command while within the repo directory.

4. **Use Python**: If the above steps don't make `yolo` directly accessible, you can run the module using Python. For example, to train YOLOv8 with Python, you would use the following command format:

```bash
python path/to/yolov8/train.py --data yourdata.yaml --cfg yolov8.yaml --weights yolov8.pt
```

By providing the correct path to the `train.py` script and necessary arguments, you can initiate training on your server. Remember to adjust the paths and file names based on your actual setup.

Please ensure that any installation and running steps are completed as per the instructions in our documentation at https://docs.ultralytics.com. If the issue persists, double-check your environment, dependencies, and paths, or provide more details so we can assist you further!

Happy training with YOLOv8 on your server! 🚀

**Comment by meeselizabeth** [comment]
@glenn-jocher Hi, thank you for your prompt response! I am actually now trying it in two different ways:

1: through the server using the python command: `CUDA_VISIBLE_DEVICES=2` mees@gpu python3 YOLOv8Model/yolov8/ultralytics/models/yolo/segment/train.py --data datasets\dataset.yaml --weights yolov8m-seg.pt --epochs 100` However, I get the error: "command not found". I figured I had not installed the requirements from the repo yet, so tried that, but then I get the error "Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'". 

2: in a singularity container using the same command. Then I get the error: "ImportError: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /.singularity.d/libs/libGLdispatch.so.0)"

I am kind of new with this. Hopefully you can help me further!

**Comment by glenn-jocher** [comment]
@meeselizabeth Hello! Let's approach each scenario separately.

### On your server:

It appears that your command has some syntactic issues. The `CUDA_VISIBLE_DEVICES` environment variable should not interfere with the command recognition, so the "command not found" message indicates another issue. Additionally, the path to your `--data` flag seems to be using backslashes (`\`), which are typical in Windows file paths. On UNIX-like systems including Linux servers, forward slashes (`/`) are used instead.

Regarding the requirements installation failure, ensure you are in the root directory of the YOLOv8 repository before you try to install the requirements. The "No such file or directory: 'requirements.txt'" error indicates that the file is not found in your current directory.

Here's how to install requirements when in the correct directory:

```bash
pip install -r requirements.txt
```

Once the requirements are successfully installed, use the corrected command to start your training:

```bash
CUDA_VISIBLE_DEVICES=2 python3 /path/to/YOLOv8Model/yolov8/train.py --data datasets/dataset.yaml --weights yolov8m-seg.pt --epochs 100
```

Make sure to provide paths that correctly point to the training script, your dataset YAML file, and use forward slashes.

### In a singularity container:

The GLIBC (GNU C Library) error suggests that the base image used in the Dockerfile might have an incompatible version of GLIBC with the binaries in your singularity container. GLIBC versions are tightly coupled to the operating system and are not backward compatible.

One potential solution is to update the Docker base image to a version that has the correct GLIBC version for your container. This can involve some trial and error to find a compatible combination.

Alternatively, you could try using a container based on a different image or building a new container specifically tailored for your environment.

Remember, resolving container-based GLIBC issues might require deep system-level troubleshooting and might be complex, especially if you're new to this. If possible, consult with someone who has experience building and managing containers for additional support.

Good luck with your YOLOv8 endeavors, and don't hesitate to reach out if you need further assistance! Keep pushing through the learning curve; it gets easier with practice. 🛠️🧠

**Comment by meeselizabeth** [comment]
@glenn-jocher Hi, thank you so much, it now works on my server! I will leave the solution for the singularity container for another time ;) 

I do have two more question:

1. Is there a way to perform hyperparameter evolution during yolov8 training? If so, how could I do that?
2. The reason I am training a yolov8 model, is because I first trained a yolov5m-seg on my data, but could not load the best.pt file. I am working on a project where I want to use the output of the segmentation model as input for another detection model. I just wanted to check whether I understood correctly that yolov5 segmentation is not supported for inference, and it is indeed necessary to retrain with a yolov8?

Looking forward to hearing from you!

**Comment by glenn-jocher** [comment]
@meeselizabeth It's great to hear that things are working on your server! Let's look at your questions:

1. **Hyperparameter Evolution in YOLOv8**: 
   Yes, YOLOv8 supports hyperparameter evolution to optimize model performance. This process is systematic and involves running several training experiments, each with a different set of hyperparameters, and then selecting the best one based on performance metrics. To perform hyperparameter evolution, you would typically set up an evolutionary algorithm that can automatically adjust hyperparameters over successive generations of training runs. We have documentation and guidelines on setting up hyperparameter tuning for YOLOv8 models. Please take a look at the documentation for a detailed explanation and instructions on how to conduct hyperparameter evolution.

2. **YOLOv5 Segmentation to YOLOv8 Migration**:
   For YOLOv5 segmentation model inference, there was indeed limited support and community effort, mostly because YOLOv5 was mainly optimized and configured for object detection tasks. However, for YOLOv8, segmentation capabilities are integrated and supported more robustly. If you want to use segmentation output as an input to another detection model, you would benefit from the improvements and support for segmentation in YOLOv8. Therefore, retraining with YOLOv8 would likely be a necessary step to get a better-supported segmentation model, as well as to utilize the model in more advanced and integrated workflows.

To ensure the smoothest experimentation and development process, make sure to refer to documentation for the most up-to-date information and best practices.

Keep up the innovative work with your project, and I'm here if you have any more questions or need further guidance. Good luck! 🌟🧑‍💻

**Comment by meeselizabeth** [comment]
@glenn-jocher Good to hear that switching to YOLOv8 was the right move, and thank you for the explanation! 

Regarding the hyperparameters: I get the errors " no directory 'yolo' " and " local variable 'ckpt_file' referenced before assignment " when using mode.tune(). This happens after the first iteration, and the results of the first iteration do get stored in user/runs/segments/tune. I've read your answer to similar questions regarding these errors on GitHub, however I do not quite understand it yet :) 

I am first defining the model as: `model= YOLO('yolov8m-seg.pt')`  and then tuning with `model.tune(data='dataset.yaml', epochs=10, iterations=300, optimizer='AdamW', plots=False, save=False, val=False` And my filepaths look like this: 

user -- YOLOv8Model -- yolov8.py
    |                           |---- dataset.yaml
    | 
    | --- runs -- segment -- train1
                                |------ train2
                                |------ tune

Where should I include the "yolo" directory? And how to solve the ckpt error? 

Hope you can help me further! Thank you :)

**Comment by glenn-jocher** [comment]
@meeselizabeth I see you're encountering errors during hyperparameter tuning with YOLOv8. Let's address these points one by one:

1. **"no directory 'yolo'" error**:
   This error message suggests that the script expects a directory named 'yolo' that it cannot find. This could be a result of the working directory not being set correctly or an issue with how the paths are handled within the code. Ensure that your environment is set up correctly and that any paths within your `tune` configuration are specified relative to the current working directory or as absolute paths.

2. **"local variable 'ckpt_file' referenced before assignment" error**:
   This error implies a variable is being used before it has been assigned a value. This could be due to conditional paths in the code that don't handle every case, or it could be indicative of a logic error or missing step prior to attempting to use `ckpt_file`.

For both errors, without providing code or specific implementation details, it's a bit challenging to diagnose the exact cause. However, these steps generally help in troubleshooting:

- Verify that the working directory is correct. Your scripts and commands should be executed from within the root of the YOLOv8 directory.
- Ensure that any reference to the 'yolo' directory is correct and that the directory exists where the code expects it to be.
- Review the control flow of your tune function to understand where 'ckpt_file' should be defined and ensure it always receives a value before being accessed.
- Double-check your configurations in `dataset.yaml` and any other relevant files to verify that paths are correctly referenced.

If these steps don't resolve your issues, it would be helpful to carefully read the documentation for any prerequisites or considerations specific to hyperparameter tuning, or to review examples from other users who have successfully performed tuning with YOLOv8.

Remember, debugging often involves revisiting the basics, so take another look at the setup and configuration details. If the errors persist, more detailed information or direct access to the code might be needed to provide specific guidance.

Keep experimenting, and feel free to reach out again! Finding solutions to such challenges is part of the learning and development process. 🛠️✨

**Comment by meeselizabeth** [comment]
@glenn-jocher Thank you! 

I added the yolo directory to $PATH and now I get the error `Permission denied: 'yolo'`. 

Some information of how I am operating:
- I have my a project folder: /autofs/gpu/user/mees/YOLOv8Model/ and in this folder I have the YOLOv8 folder, which is the cloned repository from GitHub. Then in that YOLOv8 folder I have added my datasets folder, containing dataset.yaml. And I have also added a Python file to the YOLOv8 folder named "yolov8.py"where I have written: `model = YOLO('yolov8m-seg.pt')` and `model.tune(data='dataset.yaml', epochs=10, iterations=300, etc)` 

- When I want to run the hyperparameter tuning, I run it from mees@gpu:~/YOLOv8Model/yolov8$ and I type the following command: python3 yolov8.py

- The way I added yolo to $PATH was by doing `export PATH=/.local/lib/python3.10/site-packages:$PATH` and when I do "ls -l yolo" from the path /.local/lib/python3.10/site-packages I do get a list with a lot of scripts. So I guess it is correctly added to $PATH? However, no I get the error permission denied: 'yolo'. 

Thank you for helping me, much appreciated!

**Comment by meeselizabeth** [comment]
@glenn-jocher 

Would it maybe be an idea to use the hyperparameters from the YOLOv5 I have used? Then I would not have to perform model.tune(). However, then I would have to change the hyperparameters of the YOLOv8 manually, is that possible?

Hope to hear from you!

**Comment by glenn-jocher** [comment]
@meeselizabeth Configuring your environment can be quite a complex task, especially when dealing with permissions and path-related issues. The "Permission denied: 'yolo'" error typically indicates that the current user does not have the necessary permissions to access the 'yolo' directory or execute a file within it.

Adding a directory to `$PATH` is done to allow the system to recognize executable commands found in that directory from any location in the terminal. In the context of your setup, since the 'yolo' appears to be the Python package and not a directory with executable scripts, you wouldn't typically add this to `$PATH`.

Regarding your operation setup and the current issue:
- Make sure the 'yolov8.py' Python file you've created has the necessary execute permissions.
- You don't need to add the Python package directory to `$PATH`, as Python should be able to load installed packages when you import them in scripts without altering `$PATH`.
- If you're trying to run Python scripts from a directory, make sure the user account you're using has execute and read permissions on the scripts and the directories they reside in.

If you keep encountering permission issues, here's a command that can modify permissions to grant the current user execute permissions (use this cautiously):

```bash
chmod +x /path/to/file_or_directory
```

Regarding hyperparameters:
- While transferring hyperparameters from YOLOv5 to YOLOv8 might be a practical starting point, it's important to note that these models might have different architectural features or training methodologies which could affect the effectiveness of those hyperparameters.
- Manually editing the hyperparameters of YOLOv8 is possible. You can do this by locating the hyperparameter configuration file in the YOLOv8 repository and adjusting the values according to your requirements or the YOLOv5 hyperparameters you've previously used.

Finally, if you find that tuning is necessary for YOLOv8, and you're unable to resolve the tuning issues in your current environment, it might be helpful to run this process on a local machine or a different environment where you have more control and familiarity with the setup.

As you work through these issues, remember that resolving them will strengthen your understanding of both the environment and the models. Good luck, and continue the great work! 🌟

**Comment by meeselizabeth** [comment]
@glenn-jocher Thank you for the clear explanations!

I have used the hyperparameters from the YOLOv5 and the results are pretty good. I just wanted to ask whether the (M) and (B) in the YOLOv5 and YOLOv8 results stand for Mask and Box or Macro and Best, cause they score quite similar, so I can imagine they represent the Mask and Box precision, recall and mAP - is that right? 

Also, in the YOLOv5 results the actual score is printed next to the plot, however I cannot find the actual score in the YOLOv8 results. Can I find them somewhere (instead of making a guess by looking at the plots)?

Looking forward to hearing from you again : )

**Comment by glenn-jocher** [comment]
@meeselizabeth I'm glad to hear that utilizing hyperparameters from YOLOv5 has yielded good results for you in YOLOv8!

Regarding your questions about the results:

1. In the context of YOLO results, (M) and (B) typically refer to metrics at different detection thresholds or for different classes of objects. More specifically:
   - (M) commonly stands for 'medium' objects or sometimes 'mask' if segmentation is involved, depending on the context provided by the dataset or the specific task.
   - (B) commonly stands for 'large' objects, abbreviated from 'big', or 'box' if it's related to object detection bounding boxes.

   To clarify for YOLOv8, it would depend on the specific context or the task at hand, such as object detection or segmentation, and how the metrics are defined in the output results documentation or code comments.

2. For the actual score (precision, recall, mAP, etc.) in the YOLOv8 results:
   - The detailed numerical results are typically stored in a results file or printed to the console after evaluation. This would include precision, recall, mAP scores at different IoU (Intersection over Union) thresholds, and other relevant metrics.
   - They might not be displayed directly next to the plots as in YOLOv5, but you should be able to find a summary table or a results file, usually in CSV or JSON format, in the output directory specified during training or evaluation.
   - Additionally, the performance metrics are usually logged if you are using tools like Weights & Biases or TensorBoard for experiment tracking.

If you're unable to locate the numerical scores in the outputs, ensure you're following the evaluation steps as per the documentation, where these should be detailed, and also make sure to check any logs or results directories created during training and validation.

Good luck with your continued work with the YOLO models, and don't hesitate to reach out if you have more questions or need further clarification. Your curiosity and attention to detail will serve you well in your projects! 😊📊

**Comment by meeselizabeth** [comment]
@glenn-jocher Thank you, everything is clear now and I can continue working with the YOLOv8 in my project! Your help has been very valuable :)

**Comment by glenn-jocher** [comment]
@meeselizabeth It's truly a pleasure to be of assistance! I'm thrilled to hear that you're now well-equipped to move forward with your project using YOLOv8. Should you encounter any further inquiries or challenges, please don't hesitate to reach out. Congratulations on your progress and I wish you continued success with your work. Happy detecting! 😄🎉

**Comment by meeselizabeth** [comment]
@glenn-jocher Hi, I hope everything is well! 

I am currently writing my paper, and I want to compare v5's and v8's architecture. Could you please check whether I've drawn the architectures correctly? 
<img width="519" alt="YOLOv5struct" src="https://github.com/ultralytics/ultralytics/assets/119333324/b72326d7-398b-4583-854d-21bc33770367">
<img width="511" alt="YOLOv8struct" src="https://github.com/ultralytics/ultralytics/assets/119333324/9c34d748-87f7-4f5e-bf51-7992e34df41a">
Thank you!

**Comment by dbrod772** [comment]
"'yolo' is not recognized as an internal or external command,
operable program or batch file." No methods seem to fix this error for me. I am on a Windows 11 OS. Help would be greatly appreciated!

**Comment by glenn-jocher** [comment]
@dbrod772 it sounds like the system isn't recognizing 'yolo' as an executable command. On Windows, this is often due to an incorrect PATH environment variable setting. Here are a few concise steps to troubleshoot:

1. **Ensure Python Scripts Directory in PATH**: Verify that the Python Scripts directory is properly added to your system's PATH environment variable.

2. **Activate Environment**: If you're using a virtual environment, ensure it's activated before running the command.

3. **Use Python Module Syntax**: Instead of using 'yolo', try running your script using the Python module syntax:
   ```
   python -m yolov8
   ```
   Replace 'yolov8' with the appropriate command if there is a specific module you're trying to run.

If you've tailored a 'yolo' command in a script, make sure the script's directory is included in the PATH, or use the script's full path to run it.

**Comment by glenn-jocher** [comment]
@dbrod772 The error you are encountering is due to 'yolo' not being recognized as a command. Since you're not using a virtual environment, verify that the Python executable directory is in the PATH, not just the venv scripts directory.

For YOLOv8, try replacing 'yolo' with the appropriate Python command as follows:

```bash
python path/to/yolov8/run.py task=detect mode=train --imgsz 640 --data CustomCharacterPath.yaml --epochs 1000 --batch-size 16 --name <your_experiment_name>
```

Make sure you're running this command within the directory where `run.py` is located, or provide the full path to `run.py`. Replace `<your_experiment_name>` with your desired experiment name.

**Comment by yuqilol** [comment]
Hello, I also have the same problem, the same environment (100% copy), I can use the yolo instruction on the Internet computer, but not the Internet computer can not use the yolo instruction, whether the yolo instruction needs to be networked.

**Comment by yuqilol** [comment]
My problem now is that I want to test my openvino model, but the data is trained on a computer that is not connected to the Internet, and the validation set address is also on the computer. I observe that validator.py seems to have only yolo instruction, is there any other way for me to verify the accuracy of my model

**Comment by yuqilol** [comment]
Error is as follows: - bash: / home/pengyuqi/anaconda3 / envs/yolov8 - CLS/bin/yolo: always execute: a required file not found

**Comment by glenn-jocher** [comment]
The error suggests that the `yolo` executable is missing or not properly installed in your environment. Please ensure that the Ultralytics package is correctly installed by running `pip install ultralytics` in your active environment. Once installed, try running the command `yolo` again. If the issue persists, verify the installation path and consider reinstalling the package. You can also run YOLO directly in Python using `from ultralytics import YOLO`.

**Comment by rajhlinux** [comment]
I'm getting the same error:

`F:\AI_Componets\YOLOv8>pip install ultralytics`

```
Requirement already satisfied: ultralytics in c:\users\rajh\appdata\roaming\python\python312\site-packages (8.3.56)
Requirement already satisfied: numpy>=1.23.0 in c:\python312\lib\site-packages (from ultralytics) (1.26.4)
Requirement already satisfied: matplotlib>=3.3.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (3.10.0)
Requirement already satisfied: opencv-python>=4.6.0 in c:\python312\lib\site-packages (from ultralytics) (4.10.0.84)
Requirement already satisfied: pillow>=7.1.2 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (11.0.0)
Requirement already satisfied: pyyaml>=5.3.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (6.0.2)
Requirement already satisfied: requests>=2.23.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (2.32.3)
Requirement already satisfied: scipy>=1.4.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (1.14.1)
Requirement already satisfied: torch>=1.8.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (2.5.1)
Requirement already satisfied: torchvision>=0.9.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (0.20.1)
Requirement already satisfied: tqdm>=4.64.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (4.67.1)
Requirement already satisfied: psutil in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (6.1.1)
Requirement already satisfied: py-cpuinfo in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (9.0.0)
Requirement already satisfied: pandas>=1.1.4 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (2.2.3)
Requirement already satisfied: seaborn>=0.11.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (0.13.2)
Requirement already satisfied: ultralytics-thop>=2.0.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from ultralytics) (2.0.13)
Requirement already satisfied: contourpy>=1.0.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (1.3.1)
Requirement already satisfied: cycler>=0.10 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (4.55.3)
Requirement already satisfied: kiwisolver>=1.3.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (1.4.8)
Requirement already satisfied: packaging>=20.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (24.2)
Requirement already satisfied: pyparsing>=2.3.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (3.2.0)
Requirement already satisfied: python-dateutil>=2.7 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from matplotlib>=3.3.0->ultralytics) (2.9.0.post0)
Requirement already satisfied: pytz>=2020.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from pandas>=1.1.4->ultralytics) (2024.2)
Requirement already satisfied: tzdata>=2022.7 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from pandas>=1.1.4->ultralytics) (2024.2)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from requests>=2.23.0->ultralytics) (3.4.1)
Requirement already satisfied: idna<4,>=2.5 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from requests>=2.23.0->ultralytics) (3.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from requests>=2.23.0->ultralytics) (2.3.0)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from requests>=2.23.0->ultralytics) (2024.12.14)
Requirement already satisfied: filelock in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (3.16.1)
Requirement already satisfied: typing-extensions>=4.8.0 in c:\python312\lib\site-packages (from torch>=1.8.0->ultralytics) (4.12.2)
Requirement already satisfied: networkx in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (3.4.2)
Requirement already satisfied: jinja2 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (3.1.5)
Requirement already satisfied: fsspec in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (2024.12.0)
Requirement already satisfied: setuptools in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (75.6.0)
Requirement already satisfied: sympy==1.13.1 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from torch>=1.8.0->ultralytics) (1.13.1)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from sympy==1.13.1->torch>=1.8.0->ultralytics) (1.3.0)
Requirement already satisfied: colorama in c:\python312\lib\site-packages (from tqdm>=4.64.0->ultralytics) (0.4.6)
Requirement already satisfied: six>=1.5 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from python-dateutil>=2.7->matplotlib>=3.3.0->ultralytics) (1.17.0)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\rajh\appdata\roaming\python\python312\site-packages (from jinja2->torch>=1.8.0->ultralytics) (3.0.2)

[notice] A new release of pip is available: 24.2 -> 24.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip
```

`F:\AI_Componets\YOLOv8>yolo --help`

```
'yolo' is not recognized as an internal or external command,
operable program or batch file.
```

Thanks for any advice.

**Comment by glenn-jocher** [comment]
The error indicates an issue with OpenCV (`cv2`) not being properly installed or its dependencies missing. Ensure the following to resolve it:

1. Uninstall and reinstall OpenCV with the same Python version you're using:
   ```
   pip uninstall opencv-python opencv-python-headless
   pip install opencv-python
   ```

2. Verify that your Python environment matches the architecture of your OpenCV installation (e.g., both must be 64-bit).

3. Ensure all required DLLs for OpenCV are present. You may need to install the Microsoft Visual C++ Redistributable if it's missing.

After addressing this, retry running the `yolo` command. If the issue persists, test using Python directly:  
```python
import cv2
```
If this fails, the OpenCV installation still has issues.
